test/servers/binary_framed_test.exs

```diff
@@ -75,11 +75,13 @@ defmodule Servers.BinaryFramedIntegrationTest do
   alias Thrift.TApplicationException, as: TAE
 
   def stop_server(server_pid) do
-    ref = Process.monitor(server_pid)
-    Server.Framed.stop(server_pid)
-    receive do
-      {:DOWN, ^ref, _, _, _} ->
-        :ok
+    if Process.alive?(server_pid) do
+      ref = Process.monitor(server_pid)
+      Server.Framed.stop(server_pid)
```

Comment:
@pguillory sorry you had the correct catch pattern as `GenServer.stop` has special handling for `:timeout` and `:noproc`.